packages/react-core/src/components/Page/PageHeaderToolsGroup.tsx

```diff
@@ -0,0 +1,38 @@
+import * as React from 'react';
+import styles from '@patternfly/react-styles/css/components/Page/page';
+import { css } from '@patternfly/react-styles';
+
+export interface PageHeaderToolsGroupProps extends React.HTMLProps<HTMLDivElement> {
+  /** Content rendered in the page header tools group */
+  children: React.ReactNode;
+  /** Additional classes added to the page header tools group. */
+  className?: string;
+  /** Makes the group only visible for desktop sizes (>= lg breakpoint) */
+  visibleOnLg?: boolean;
+  /** Hides the group for desktop sizes (< lg breakpoint) */
+  hiddenOnLg?: boolean;
+  /** Hides the group for mobile sizes (< md breakpoint). Typically used on a user dropdown */
+  hiddenOnSm?: boolean;
+}
+
+export const PageHeaderToolsGroup: React.FunctionComponent<PageHeaderToolsGroupProps> = ({
+  children,
+  className,
+  hiddenOnLg,
+  visibleOnLg,
+  hiddenOnSm,
+  ...props
+}: PageHeaderToolsGroupProps) => (
+  <div
+    className={css(
+      styles.pageHeaderToolsGroup,
+      className,
+      hiddenOnLg && styles.modifiers.mobile,
```

Comment:
If we're going to structure this so that we offer header toolbar groups and items that require the user specify `visibleOnLg` for items that are only visible on desktop, `hiddenOnSm` for items that are hidden on mobile, etc, then I wonder if we should drop the current modifiers and use a hidden/visible mixin that would give us a set of styles that allow a user to specify `pf-m-hidden[-on-{breakpoint}]` and `pf-m-visible[-on-{breakpoint}]` on any header-tools-group or header-tools-item. In this PR, users can only specify `hiddenOnLg `, `visibleOnLg`, and `hiddenOnSm`.

Then users will have full control over hiding/showing things at any breakpoint. The downside to the current approach (whether we include the mixin or not) is that the responsibility is on the user to hide/show header toolbar items appropriately. The alternative to that would be to have something like 3 props on the header tools component, `mobileDropdown`, `desktopActions`, and `userMenu` or something like that, and we put the contents of those props in the appropriate classes and control the mobile, desktop, and user menu breakpoints where that content is hidden/shown.

The former seems like the better approach, but could lead to inconsistency in implementation. @mcarrano wdyt? If we're OK allowing users to hide/show any header toolbar items/groups at any breakpoint, but we show the proper way to do it in the demos and could have a masthead/header design guideline that goes into more detail, I'd like to drop the existing classes and add the full hidden/visible breakpoint classes to header toolbar groups and items.

We could keep the existing classes and add the hidden/visible classes on top, but that seems like it might be confusing to have both, and the prop names that enable the current classes seem like they would overlap.

Reply:
Yes, I was going back and forth with the prop names, initially I named them something similar to your mentioned alternative before landing on this (as it also seemed familiar with what we have in other components). 
I guess if we wanted to we could streamline this even more and provide a way for users to define specific things and we can lay it out for them:
```
<PageHeaderTools 
   mobileDropdown={<Dropdown />}
   desktopActions={
      { component: <NotificationBadge />, isSelected: false }
      { component: <Button><CogIcon /></Button>, isSelected: true }
   }
   userDropdown={
      <Dropdown />
   }
   avatar={<Avatar />}
>
   Other components here that should come before?
</PageHeaderTools>
```

Maybe this for ease-of-use, but still allow users to use the <PageHeaderTools> component without those convenience props and define the content themselves using visibility mixins?

Reply:
Looking at the way it's done in master now, it looks like we import the accessibility utility classes and it's up to users to apply hidden/visible classes on their own. It seems logical then to bake this hidden/visible behavior into the page header tools, and give users the ability to hide/show header elements at any breakpoint.